pkg/db-repo/repository.go

```diff
@@ -246,7 +299,14 @@ func (r *repository) Query(ctx context.Context, qb *QueryBuilder, result interfa
 		db = db.Limit(qb.page.limit)
 	}
 
-	err := db.Find(result).Error
+	err := r.checkResultModel(result)
```

Comment:
I suggest moving this check to the beginning of the file, like for the other operations.
Prevents doing unnecessary stuff :D

Reply:
Done